src/main/java/quotify_app/adapters/currentprice/CurrentPricePresenter.java

```diff
@@ -0,0 +1,64 @@
+package quotify_app.adapters.currentprice;
+
+import quotify_app.adapters.ViewManagerModel;
+import quotify_app.app.ApplicationState;
+import quotify_app.usecases.currentprice.CurrentPriceOutputBoundary;
+
+/**
+ * The Presenter for the Current Price use case.
+ * Processes the output from the Interactor and updates the ViewModel or handles navigation.
+ */
+public class CurrentPricePresenter implements CurrentPriceOutputBoundary {
+
+    private final ViewManagerModel viewManagerModel;
+    private final CurrentPriceViewModel currentPriceViewModel;
+
+    /**
+     * Initializes the CurrentPricePresenter with the given ViewManagerModel and CurrentPriceViewModel.
+     *
+     * @param viewManagerModel      the model managing view transitions.
+     * @param currentPriceViewModel the ViewModel for the Current Price View.
+     */
+    public CurrentPricePresenter(ViewManagerModel viewManagerModel, CurrentPriceViewModel currentPriceViewModel) {
+        this.viewManagerModel = viewManagerModel;
+        this.currentPriceViewModel = currentPriceViewModel;
+    }
+
+    @Override
+    public void presentGoToLogin() {
+        viewManagerModel.setState("log in");
+        viewManagerModel.firePropertyChanged();
+    }
+
+    @Override
+    public void presentGoToSignup() {
+        viewManagerModel.setState("sign up");
+        viewManagerModel.firePropertyChanged();
+    }
```

Comment:
Rather than having a log in and signup option on your view, maybe have a back to landing page button, which when clicked clears all data that has been stored about the property.

Reply:
Done, it shows a button called Home now, when a user is in guest mode and when user is logged in it shows a user profile button.